front/pages/api/w/[wId]/webhook_sources/views/[viewId]/index.ts

```diff
@@ -115,4 +149,37 @@ async function handler(
   }
 }
 
+async function propagateSystemViewNameChange(
+  auth: Authenticator,
+  webhookSourceView: WebhookSourcesViewResource,
+  newName: string
+): Promise<Result<undefined, DustError<"unauthorized">>> {
+  // Propagate changes to system view and all space views
+  const systemView =
+    await WebhookSourcesViewResource.getWebhookSourceViewForSystemSpace(
+      auth,
+      webhookSourceView.webhookSourceSId
+    );
+
+  const isSystemView = systemView?.sId === webhookSourceView.sId;
+
+  if (isSystemView) {
+    // This is the system view, update all space views with the same webhook source
+    const allViews = await WebhookSourcesViewResource.listByWebhookSource(
+      auth,
+      webhookSourceView.webhookSourceId
+    );
+
+    for (const view of allViews) {
+      if (view.sId !== webhookSourceView.sId) {
+        const viewUpdateResult = await view.updateName(auth, newName);
+        if (viewUpdateResult.isErr()) {
+          return viewUpdateResult;
+        }
+      }
+    }
+  }
```

Comment:
We might want a transaction here to avoid views being created in the meantime, or at least start with updating systemView